src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+            {
+                string columnNameForComponent = column;
+
+                if (_metadataProvider.TryGetExposedColumnName(entityName, column, out string? mappedColumnAlias) && !string.IsNullOrEmpty(mappedColumnAlias))
+                {
+                    columnNameForComponent = mappedColumnAlias;
+                }
+
+                if (sourceDefinition.Columns.TryGetValue(columnNameForComponent, out ColumnDefinition? columnDef))
+                {
+                    OpenApiSchema parameterSchema = new()
+                    {
+                        Type = columnDef is not null ? SystemTypeToJsonDataType(columnDef.SystemType) : string.Empty
+                    };
+
+                    OpenApiParameter openApiParameter = new()
+                    {
+                        Required = true,
+                        In = ParameterLocation.Path,
+                        Name = $"{columnNameForComponent}",
+                        Schema = parameterSchema
+                    };
+
+                    parameters.Add(openApiParameter);
+                    string pkComponent = $"/{columnNameForComponent}/{{{columnNameForComponent}}}";
+                    pkComponents.Append(pkComponent);
+                }
+            }
+
+            return new(pkComponents.ToString(), parameters);
+        }
+
+        /// <summary>
+        /// Determines whether the database object has an autogenerated primary key
+        /// used to distinguish which request and response objects should reference the primary key.
+        /// </summary>
+        /// <param name="sourceDefinition">Database object metadata.</param>
+        /// <returns>True, when the primary key is autogenerated. Otherwise, false.</returns>
+        private static bool SourceContainsAutogeneratedPrimaryKey(SourceDefinition sourceDefinition)
+        {
+            bool sourceObjectHasAutogeneratedPK = false;
+            // Create primary key path component.
+            foreach (string column in sourceDefinition.PrimaryKey)
+            {
+                string columnNameForComponent = column;
+
+                if (sourceDefinition.Columns.TryGetValue(columnNameForComponent, out ColumnDefinition? columnDef) && columnDef is not null && columnDef.IsAutoGenerated)
+                {
+                    sourceObjectHasAutogeneratedPK = true;
+                }
+            }
+
+            return sourceObjectHasAutogeneratedPK;
+        }
+
+        /// <summary>
+        /// Evaluates a database object's fields to determine whether a request body is required.
+        /// A body is required when any one field
+        /// - is auto generated
+        /// - has a default value
+        /// - is nullable
+        /// </summary>
+        /// <param name="sourceDef">Database object's source metadata.</param>
+        /// <returns>True, when a body should be generated. Otherwise, false.</returns>
+        private static bool IsRequestBodyRequired(SourceDefinition sourceDef)
```

Comment:
```suggestion
        private static bool IsRequestBodyRequiredForMutation(SourceDefinition sourceDef)
```

Reply:
Thanks for the suggestion, I'm going to refrain from using this name change since the term 'mutation' is attributed to GraphQL. I am going to update the comment to reflect the HTTP verbs with which a body is typically included: POST, PUT, PATCH.

Reply:
sounds good.

Reply:
updated comment.